libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationSymbolManager.java

```diff
@@ -26,8 +27,12 @@
   }
 
   void addDestinationMarkerFor(@NonNull Point position) {
+    if (destinationSymbol != null) {
+      symbolManager.delete(destinationSymbol);
+      mapMarkersSymbols.remove(destinationSymbol);
+    }
     SymbolOptions options = createSymbolOptionsFor(position);
-    createSymbolFrom(options);
+    destinationSymbol = createSymbolFrom(options);
   }
 
   void addCustomSymbolFor(@NonNull SymbolOptions options) {
```

Comment:
do we also want to save the custom symbol to `destinationSymbol` here and so we can clear it later?